source/nodejs/adaptivecards/src/card-elements.ts

```diff
@@ -3457,23 +3431,10 @@ export class ChoiceSetInput extends Input {
 
             input.onchange = () => {
                 this.valueChanged();
-                if (choice.inlineAction instanceof ToggleVisibilityAction) {
-                    let toggleVisibilityAction = choice.inlineAction as ToggleVisibilityAction;
-
-                    let newTargetElements: { [key: string]: any } = {};
-                    for (let elementId of Object.keys(toggleVisibilityAction.targetElements)) {
-                        let targetElement = toggleVisibilityAction.targetElements[elementId];
-
-                        if (input.type !== "radio") {
-                            targetElement = input.checked;
-                        }
-
-                        newTargetElements[elementId] = targetElement;
-                    }
 
-                    toggleVisibilityAction.targetElements = newTargetElements;
-                    toggleVisibilityAction.setParent(this);
-                    toggleVisibilityAction.execute();
+                if (choice.inlineAction) {
```

Comment:
You removed the check that `inlineAction` is of type `ToggleVisibilityAction` which is fine, that said if `inlineAction` is set in code it can be set to another type of action given that `inlineAction` is defined as being of type `Action`. Maybe change the type of the field to `ToggleVisibilityAction`, given that's all it's supposed to be?

Reply:
Also, you do have to check for `input.checked` here but only when the input in in single select expanded mode. Otherwise what's going to happen is the `inlineAction` of both the newly selected radio button and that of the one that isn't selected anymore are going to be triggered. Given that the `inlineAction` as defined in the card is meant to be executed only when the radio button is checked, that's going to yield the wrong result.